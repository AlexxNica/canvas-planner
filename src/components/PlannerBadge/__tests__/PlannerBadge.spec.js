import React from 'react';
import { shallow } from 'enzyme';
import 'instructure-ui/lib/themes/canvas';
import { PlannerBadge } from '../index';

it('renders the base component', () => {
  const wrapper = shallow(
    <PlannerBadge>
      <span>Hello</span>
    </PlannerBadge>
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the count when it is provided as a prop', () => {
  const wrapper = shallow(
    <PlannerBadge count={40}>
      <span>Hello</span>
    </PlannerBadge>
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('span').length).toEqual(4);
});

it('does not render the count when it is zero', () => {
  const wrapper = shallow(
    <PlannerBadge count={0}>
      <span>Hello</span>
    </PlannerBadge>
  );
  expect(wrapper.find('PresentationContent').length).toEqual(0);
  expect(wrapper.find('span').length).toEqual(3);
});
