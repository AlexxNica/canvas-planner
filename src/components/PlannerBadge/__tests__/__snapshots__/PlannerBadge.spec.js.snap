// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the base component 1`] = `
<span>
  <span>
    <span>
      Hello
    </span>
  </span>
</span>
`;

exports[`renders the count when it is provided as a prop 1`] = `
<span>
  <span>
    <PresentationContent
      as="span"
    >
      40
    </PresentationContent>
  </span>
  <span>
    <span>
      Hello
    </span>
  </span>
</span>
`;
